packages/frontend/src/state/lp/hooks.ts

```diff
@@ -68,6 +69,62 @@ export const useOpenPositionDeposit = () => {
   return openPositionDeposit
 }
 
+// Close position with flashloan
+export const useClosePosition = () => {
+  const address = useAtomValue(addressAtom)
+  const controllerHelperContract = useAtomValue(controllerHelperHelperContractAtom)
+  const controllerContract = useAtomValue(controllerContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const getVault = useGetVault()
+  const getPosition = useGetPosition()
+  const getQuote = useGetQuote()
+  const closePosition = useAppCallback(async (vaultId: number, onTxConfirmed?: () => void) => {
+    const vaultBefore = await getVault(vaultId)
+    const uniTokenId = vaultBefore?.NFTCollateralId 
+    const position = await getPosition(uniTokenId)
+
+    if (
+      !controllerContract ||
+      !controllerHelperContract ||
+      !address ||
+      !position ||
+      !vaultBefore ||
+      !vaultBefore.shortAmount
+    )
+      return
+
+      const shortAmount = fromTokenAmount(vaultBefore.shortAmount, OSQUEETH_DECIMALS)
+      const debtInEthPromise = getDebtAmount(shortAmount)
+      const limitEthPromise = getQuote(shortAmount, true)
+      const [debtInEth, limitEth] = await Promise.all([debtInEthPromise, limitEthPromise])
+      const collateralToFlashloan = debtInEth.multipliedBy(COLLAT_RATIO)
+
+    const flashloanCloseVaultLpNftParam = {
+      vaultId: vaultId,
+      tokenId: uniTokenId,
+      liquidity: position.liquidity,
+      liquidityPercentage: fromTokenAmount(1, 18).toFixed(0),
```

Comment:
IMO for a more flexible API this should be flexible

Reply:
users could decide to close 50% of their LP to pay off 75% their debt as an example